cmd/kubeadm/app/util/kubeconfig/kubeconfig_test.go

```diff
@@ -345,3 +346,182 @@ func TestHasCredentials(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterFromKubeConfig(t *testing.T) {
+	tests := []struct {
+		name      string
+		config    *clientcmdapi.Config
+		expected  string
+		expected1 *clientcmdapi.Cluster
+	}{
+		{
+			name: "cluster is empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "kubernetes",
+			},
+			expected:  "",
+			expected1: nil,
+		},
+		{
+			name: "cluster and currentContext are not empty",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"foo": {AuthInfo: "foo", Cluster: "foo"},
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expected: "foo",
+			expected1: &clientcmdapi.Cluster{
+				Server: "http://foo:8080",
+			},
+		},
+		{
+			name: "cluster is not empty and currentContext is not in contexts",
+			config: &clientcmdapi.Config{
+				CurrentContext: "foo",
+				Contexts: map[string]*clientcmdapi.Context{
+					"bar": {AuthInfo: "bar", Cluster: "bar"},
+				},
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"foo": {Server: "http://foo:8080"},
+					"bar": {Server: "https://bar:16443"},
+				},
+			},
+			expected:  "",
+			expected1: nil,
+		},
+	}
+	for _, rt := range tests {
+		t.Run(rt.name, func(t *testing.T) {
+			got, got1 := GetClusterFromKubeConfig(rt.config)
+			if got != rt.expected {
+				t.Errorf("GetClusterFromKubeConfig() got = %v, expected %v", got, rt.expected)
```

Comment:
the go test output prints the name of the test function so it can be just 
```suggestion
				t.Errorf("got cluster name = %s, expected %s", got, rt.expected)
```